src/main/java/formflow/library/ScreenController.java

```diff
@@ -434,20 +436,32 @@ ModelAndView deleteSubflowIteration(
    * @return the screen template with model data, returns error page on error
    */
   @GetMapping("{flow}/{screen}/navigation")
-  RedirectView navigation(
+  ModelAndView navigation(
       @PathVariable String flow,
       @PathVariable String screen,
       HttpSession httpSession
-  ) {
+  ) throws Exception {
     var currentScreen = getScreenConfig(flow, screen);
     log.info("navigation: flow: " + flow + ", screen: " + screen);
     if (currentScreen == null) {
-      return new RedirectView("/error");
+      errorScreen(HttpMethod.GET.toString(), String.format("/flow/%s/%s", flow, screen), HttpStatus.NOT_FOUND);
     }
     String nextScreen = getNextScreenName(httpSession, currentScreen, null);
 
     log.info("navigation: flow: " + flow + ", nextScreen: " + nextScreen);
-    return new RedirectView("/flow/%s/%s".formatted(flow, nextScreen));
+    return new ModelAndView(new RedirectView("/flow/%s/%s".formatted(flow, nextScreen)));
+  }
+
+  private static void errorScreen(String method, String url, HttpStatus httpStatus) throws Exception {
+    HttpHeaders headers = new HttpHeaders();
+
+    if (httpStatus.isSameCodeAs(HttpStatus.NOT_FOUND)) {
+      throw new NoHandlerFoundException(method, url, headers);
+    }
+
+    if (httpStatus.isSameCodeAs(HttpStatus.BAD_REQUEST)) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+    }
```

Comment:
This code update is great!  My one thought is should we have it thrown a generic exception if it's not NOT_FOUND or BAD_REQUEST?  Not sure what it would be, but it seems like we should have some sort of generic reporting in that scenario so we don't have a silent failure.

Reply:
Ya, I'll add a default exception that passes what it's given

Reply:
Actually, `ResponseStatusException` is a good default. Since that's only a one liner, I refactored the places that want to throw a `BAD_REQUEST`, to throw in place. Then I refactored `errorScreen` to `throwNotFoundError` and it doesn't have control logic.